src/workflows-app/RunDetails.js

```diff
@@ -151,6 +156,30 @@ export const BaseRunDetails = (
     [signal, workspaceId]
   );
 
+  const renderInProgressElement = (workflow) => {
+    if (workflow.status === 'Running') {
+      return span({ style: { fontStyle: 'italic' } }, ['In progress - ']);
+    }
+  };
+
+  const calculateTotalCost = (callObjects) => {
+    let total = 0;
+    Object.values(callObjects).forEach((call) => {
+      if (!call[0].taskStartTime) {
+        total += 0;
+      } else if (call[0].taskEndTime) {
+        total += getTaskCost(call[0].vmCostUsd, call[0].taskStartTime, call[0].taskEndTime);
+      } else {
+        total += getTaskCost(call[0].vmCostUsd, call[0].taskStartTime);
+      }
+    });
+    return total;
+  };
+
+  const taskCostTotal = useMemo(() => {
```

Comment:
Is the `useMemo` doing anything here? It looks like it would only be called once per page-load anyway?

Reply:
Yeah, so from my understanding this is a way of caching the result on each re-render. It'll only run the function if its dependency, `callObjects` changes. That way we're not running this function on every re-render/page-refresh, only when necessary.

Reply:
Also in response to shards, updating the code per your suggestion didn't seem to work? It looks like `shards` are nested as sub-workflows, so when they appear at the top level of the page, they give an option to `View Subworkflow` until you navigate down and eventually get the cost of that task (next slice will focus on bubbling this cost up each level) 
<img width="1404" alt="image" src="https://github.com/DataBiosphere/terra-ui/assets/68349264/ba929a31-6144-4112-aa9f-652cbdfa480c">
<img width="1404" alt="image" src="https://github.com/DataBiosphere/terra-ui/assets/68349264/ce2d4ba8-0976-4213-82df-b5fb7c889efb">